packages/thirdweb/src/wallets/in-app/core/users/getUser.ts

```diff
@@ -76,15 +80,25 @@ export async function getUser({
   } else if (id) {
     url.searchParams.set("queryBy", "id");
     url.searchParams.set("id", id);
+  } else if (externalWalletAddress) {
+    url.searchParams.set("queryBy", "externalWalletAddress");
+    url.searchParams.set("externalWalletAddress", externalWalletAddress);
   } else {
     throw new Error(
-      "Please provide a walletAddress, email, phone, or id to query for users.",
+      "Please provide a walletAddress, email, phone, id, or externalWalletAddress to query for users.",
     );
   }
 
   const clientFetch = getClientFetch(client);
 
-  const res = await clientFetch(url.toString());
+  const res = await clientFetch(url.toString(), {
+    headers: {
+      ...(ecosystem?.id ? { "x-ecosystem-id": ecosystem.id } : {}),
+      ...(ecosystem?.partnerId
+        ? { "x-ecosystem-partner-id": ecosystem.partnerId }
+        : {}),
```

Comment:
you dont need to do this! just pass ecosystem to `getClientFetch` :) 

actually just did it in my other PR, though i didn't add externalWalletAddress support. just merge my change and you'll get the headers for free